include/rmm/device_scalar.hpp

```diff
@@ -73,15 +73,32 @@ class device_scalar {
     return host_value;
   }
 
+  /**---------------------------------------------------------------------------*
+   * @brief Copies the value from hostto device.
+   *
+   * @param host_value The value of the scalar after synchronizing its stream
+   *---------------------------------------------------------------------------**/
+  void set_value(T host_value) {
+    auto status = cudaMemcpyAsync(buff.data(), &host_value, sizeof(T),
+                                  cudaMemcpyDefault, buff.stream());
+    if (cudaSuccess != status) {
+      throw std::runtime_error{"Device memcpy failed."};
+    }
+    status = cudaStreamSynchronize(buff.stream());
```

Comment:
I think we need the option to not synchronize the stream when we copy. So perhaps we need `set_value_async` also, which takes a stream.